packages/ai-chat/src/common/chat-session-naming-service.ts

```diff
@@ -0,0 +1,122 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import {
+    Agent,
+    CommunicationRecordingService,
+    getTextOfResponse,
+    LanguageModelRegistry,
+    LanguageModelRequest,
+    LanguageModelRequirement,
+    PromptService,
+    PromptTemplate
+} from '@theia/ai-core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { ChatSession } from './chat-service';
+import { generateUuid } from '@theia/core';
+
+const CHAT_SESSION_NAMING_PROMPT = {
+    id: 'chat-session-naming-service-prompt',
+    template: '{{!-- Made improvements or adaptations to this prompt template? We\'d love for you to share it with the community! Contribute back here: ' +
+        'https://github.com/eclipse-theia/theia/discussions/new?category=prompt-template-contribution --}}\n\n' +
+        'Provide a short and descriptive name for the given AI chat conversation of an AI-powered tool based on the conversation below.' +
+        'The purpose of the name is for users to recognize the chat conversation easily in a list of conversations.' +
+        'If the chat message contains @<some-name>, this means that the user addressed a specific chat agent.' +
```

Comment:
Does this help in your experience?
```suggestion
        'If the chat message contains @<some-name>, this means that the user addressed a specific chat agent.' +
```

Reply:
I did some vibe testing and it indeed doesn't seem to have a great impact actually. I'll remove it. But I added better line breaks and mention that chat conversation names should start with an upper case.